### PR TITLE
feat: add taqueria docker image

### PR DESCRIPTION
**Pull request by GImbrailo**
This change adds a docker image for `taqueria` to the GitHub registry enabling us to use `taq` with docker. Because the `ligo` and `flextesa` plugins use docker themselves we end up with a `docker-in-docker` scenario where we need to mount volumes and set the image `workdir` correctly for those plugins to work. 

Here are some examples of how to use the image:

```shell
docker run -it --rm \
  --volume "$(pwd):$(pwd)" \
  --volume "/var/run/docker.sock:/var/run/docker.sock" \
  --workdir "$(pwd)" \
  ghcr.io/ecadlabs/taqueria:1106-merge-4a353b8 \
  taq init -p test-project
```

```shell
docker run -it --rm \
  --volume "$(pwd):$(pwd)" \
  --volume "/var/run/docker.sock:/var/run/docker.sock" \
  --workdir "$(pwd)" \
  ghcr.io/ecadlabs/taqueria:1106-merge-4a353b8 \
  taq compile
```

### CODE REVIEW

File: .github/workflows/taqueria-docker-publish.yml [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+        id: build-bin
+        shell: bash
+        env:
+          TAQ_VERSION: ${{ github.ref_name }}
+          DENO_DIR: "./deno"
+        run: |
+          BUILD=`git rev-parse --short "$GITHUB_SHA"`
+          TAQ_VERSION="${TAQ_VERSION/\//-}"
+          deno compile --output taq --allow-run --allow-write --allow-read --allow-env --allow-net --import-map ./import_map.json --no-prompt --target $DENO_TARGET index.ts --quickstart "`cat quickstart.md`" --setBuild "$BUILD" --setVersion "$TAQ_VERSION" --lock ./deno-lock.json
+          ./taq init -p ./test_project
```
Comment (alexzbusko) [inline .github/workflows/taqueria-docker-publish.yml:57]: This is usually a test in bash after the compile step right? Is there a reason that this isn't like `[[ $(./taq init -p ./test_project) == "Project taq'ified!" ]] `?

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,15 @@
+
+FROM alpine:3.16
+FROM node:16
+
+# Set the DENO_DIR environment variable to controll where the cache is built
```
Comment (alexzbusko) [inline Dockerfile:5]: Extra l in comment

File: .github/workflows/taqueria-docker-publish.yml (hunk 1 above)
Comment (GImbrailo) [reply .github/workflows/taqueria-docker-publish.yml:54]: Thanks @alexzbusko, I reverted the change to remove that test

**Review by alexzbusko — APPROVED** [review]

**Comment by alexzbusko** [comment]
@GImbrailo one thing I noticed is that the analytics prompt for most commands will cause an error if the user answers yes like this:
```
Help improve Taqueria by sharing anonymous usage statistics in accordance with the privacy policy? (Y/n) Y
error: Uncaught (in promise) Error
export const toPromise = (f)=>pipe(f, mapRej((taqErr)=>new TaqError.E_TaqError(taqErr)), promise);
                                                       ^
    at file:///home/runner/work/taqueria/taqueria/taqueria-utils/taqueria-utils.ts:134:56
    at call (https://cdn.jsdelivr.net/gh/fluture-js/Fluture@14.0.0/dist/module.js:501:29)
    at Transformation.MapRejTransformation$rejected (https://cdn.jsdelivr.net/gh/fluture-js/Fluture@14.0.0/dist/module.js:1853:37)
    at Transformation.transformationHandler [as rejected] (https://cdn.jsdelivr.net/gh/fluture-js/Fluture@14.0.0/dist/module.js:1105:19)
    at rejected (https://cdn.jsdelivr.net/gh/fluture-js/Fluture@14.0.0/dist/module.js:977:27)
    at Interpreter.Reject$interpret [as _interpret] (https://cdn.jsdelivr.net/gh/fluture-js/Fluture@14.0.0/dist/module.js:890:3)
    at drain (https://cdn.jsdelivr.net/gh/fluture-js/Fluture@14.0.0/dist/module.js:1047:25)
    at settle (https://cdn.jsdelivr.net/gh/fluture-js/Fluture@14.0.0/dist/module.js:970:15)
    at rejected (https://cdn.jsdelivr.net/gh/fluture-js/Fluture@14.0.0/dist/module.js:977:5)
    at EncaseP$rej (https://cdn.jsdelivr.net/gh/fluture-js/Fluture@14.0.0/dist/module.js:1275:7)
```

**Comment by hu3man** [comment]
Hey @GImbrailo, wanted to check in on where we left off with this before you take off

**Pull request closed without merging**